### PR TITLE
OpenTSDB: Convert the OpenTSDB Query Editor from Angular to React

### LINKED ISSUE #50639: Remove Angular dependencies from OpenTSDB
Migrate the annotations and query builder in OpenTSDB from angular to React and remove all angular code in the datasource.

### PR DESCRIPTION
**Pull request by bohandley**
This PR is to convert the OpenTSDB query editor to React. Fixes https://github.com/grafana/grafana/issues/50639

The [plan](https://docs.google.com/document/d/1HQ5xb8VZ0EKMgLtkacem1x5tJWjrD2Zs5lk81pz38Eo/edit#) is to change as little as possible but there may be small visual changes and component substitutions.

Testing

Compare this branch with the main branch. You can run both locally or you can run one docker version and this branch locally. In the custom.ini file you can change the http_port to 3001and the login_cookie_name so that you can switch between versions.

Run `make devenv sources=opentsdb` to have two provisioned opentsdb data sources with which to test. 

Manual test scenarios 
1. Run a query in Explore/Dashboards
- Select a metric, `cpu`
- Downsample
- The “fill” component is available for v2.3
- Add, edit, delete a filter for v2.3
- Add, edit, delete tags for v2.2 or less
2. Loading empty query in Explore/Dashboards
3. Loading existing query in Explore/Dashboards
4. Migrate a query from main to this branch. Create a query as shown above
- Switch branches and check that the query populates the query editor
5. Adding, removing, editing metric names (with autocomplete)
6. Check that the Select component used in the React version behaves similarly to the autocomplete in Angular
7. Adding, removing, editing tags/filters
8. Create variables and use them in metric names and function parameters
9. Error handling 
- Tags cannot be used with filters
- Two tags keys cannot be used

Notable changes
1. The metric input is changed to a select input to support autocomplete.
2. The suggested metrics are loaded once and filtered as opposed to before where they were loaded on every keystroke.
    -remove the autocomplete input
    -use a searchable select
4. The OpenTsdbQuery type has been updated to handle things that were previously defined in the target.
https://github.com/grafana/grafana/blob/7afd6bfa397e98b244dbcd857cd5ffe9b6102658/public/app/plugins/datasource/opentsdb/types.ts#L3-L20
5. Remove all angular code I come across, eg https://github.com/grafana/grafana/blob/c0c4409abc85276d49b81e8aecbfc7467a2c6362/public/app/plugins/datasource/opentsdb/datasource.ts#L540-L551
6. `suggestTagValues()` is called once now and filtered in a searchable select instead of being loaded every keystroke in the angular autocomplete (same as suggest metrics)
7. Small change to `suggestTagKeys()` in pass in query instead of metric to handle type issue.

### CODE REVIEW

File: .betterer.results [hunk 1]
```diff
@@ -7229,6 +7229,10 @@ exports[`better eslint`] = {
       [0, 0, 0, "Unexpected any. Specify a different type.", "3"],
       [0, 0, 0, "Unexpected any. Specify a different type.", "4"]
     ],
+    "public/app/plugins/datasource/opentsdb/components/MetricSection.tsx:5381": [
+      [0, 0, 0, "Unexpected any. Specify a different type.", "0"],
+      [0, 0, 0, "Unexpected any. Specify a different type.", "1"]
+    ],
```
Comment (joshhunt) [inline .betterer.results:7235]: Please do not add new exceptions into the betterer file. Our goal is to remove this tool completely, which we cannot do if we continue to grow it.

Please either type your code correctly, or use `// eslint-ignore-next-line` if following the lint rule is absolutely not possible.

Comment (bohandley) [reply .betterer.results]: Thank you Josh, sorry about that! I got a result from betterer where is said I had used 'any' at the top of a file and I thought that was a false positive. It was not, I found it and used the correct type.

File: public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  query,
+  range,
+  queries,
+}: OpenTsdbQueryEditorProps) {
+  const styles = useStyles2(getStyles);
+
+  // errors will be updated so they need to be state
+  // They are only displayed for tags, filters and downsampling.
+  // There is evidence that metrics had errors at one point,
+  // but there is no evidence it is being handled in the fromt end,
```
Comment (itsmylife) [inline public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx:33]:
```suggestion
  // but there is no evidence it is being handled in the frontend,
```

File: public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  // just errors delivered from the backend
+  const [errors /*,setErrors*/] = useState<OpenTsdbErrors>({});
```
Comment (itsmylife) [inline public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx:35]:
```suggestion
  const [errors] = useState<OpenTsdbErrors>({});
```

Comment (itsmylife) [reply public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx:35]: It looks better I think. Feel free to ignore if that this necessary

File: public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+  // Xaggregators: any;
```
Comment (itsmylife) [inline public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx:37]: Do we need those comments starting with `X`?

**Review by itsmylife — COMMENTED** [review]
Added some comments the rest looks fine. I'd like to take a second look after your input.

File: public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx (hunk 4 above)
Comment (bohandley) [reply public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx:37]: Nope! that is just my list in the code of things that I have handled and I use them as a checkpoint for regressions. I will remove them in the future.

**Review by itsmylife — APPROVED** [review]
🚀

**Pull request merged**